demos/instr.cc

```diff
@@ -114,3 +114,53 @@ uint64_t ReadLatency(const void *memory) {
   LFence();  // Necessary for x86 MSVC.
   return result;
 }
+
+#ifdef __GNUC__
+__attribute__((noinline))
+#if defined(__x86_64__) || defined(_M_X64)
+void UnwindStackAndSlowlyReturnTo(const void *address) {
+  asm volatile(
+      "addq $8, %%rsp\n"
+      "popstack:\n"
+      "addq $8, %%rsp\n"
+      "cmpq %0, (%%rsp)\n"
+      "jnz popstack\n"
+      "clflush (%%rsp)\n"
+      "mfence\n"
+      "lfence\n"
+      "ret\n"::"r"(address));
+#elif defined(__i386__) || defined(_M_IX86)
```

Comment:
I think we can skip x86 32-bit.

Reply:
Is that our global policy that we don't care about 32bit systems?

Reply:
If we don't care about 32bit systems we should remove much more branches than just this one.

Reply:
Yes, I feel pretty comfortable saying we shouldn't worry about making each demo work for 32-bit. Open to pushback on that.

I can imagine some future CPU bug that only manifests if we're using non-native-bitness instructions, but we'll cross that bridge when we come to it.

All that said, I'm fine if you want to leave it in for this PR.